tests/ImageSharp.Tests/MemoryAllocatorValidator.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Threading;
+using SixLabors.ImageSharp.Diagnostics;
+using Xunit;
+
+namespace SixLabors.ImageSharp.Tests
+{
+    public static class MemoryAllocatorValidator
+    {
+        private static readonly AsyncLocal<TestMemoryDiagnostics> LocalInstance = new();
+
+        public static bool MonitoringAllocations => LocalInstance.Value != null;
+
+        static MemoryAllocatorValidator()
+        {
+            MemoryDiagnostics.MemoryAllocated += MemoryDiagnostics_MemoryAllocated;
+            MemoryDiagnostics.MemoryReleased += MemoryDiagnostics_MemoryReleased;
+        }
+
+        private static void MemoryDiagnostics_MemoryReleased()
+        {
+            TestMemoryDiagnostics backing = LocalInstance.Value;
+            if (backing != null)
+            {
+                backing.TotalRemainingAllocated--;
+            }
+        }
+
+        private static void MemoryDiagnostics_MemoryAllocated()
+        {
+            TestMemoryDiagnostics backing = LocalInstance.Value;
+            if (backing != null)
+            {
+                backing.TotalAllocated++;
+                backing.TotalRemainingAllocated++;
+            }
+        }
+
+        public static TestMemoryDiagnostics MonitorAllocations()
+        {
+            var diag = new TestMemoryDiagnostics();
+            LocalInstance.Value = diag;
+            return diag;
+        }
+
+        public static void StopMonitoringAllocations() => LocalInstance.Value = null;
+
+        public static void ValidateAllocations(int expectedAllocationCount = 0)
+            => LocalInstance.Value?.Validate(expectedAllocationCount);
+
+        public class TestMemoryDiagnostics : IDisposable
+        {
+            public int TotalAllocated { get; set; }
+
+            public int TotalRemainingAllocated { get; set; }
+
+            public void Validate(int expectedAllocationCount)
```

Comment:
`expectedAllocationCount` seems to be always 0.

Reply:
In our current tests your are right, but I was envisioning use cases where we might want to want to validate before all buffers are returned and this is to allow for testing a specific/expected number of buffers are currently allocated still.